core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/SdkInternalExecutionAttribute.java

```diff
@@ -34,11 +35,23 @@
 
     /**
      * The key to store the {@link CompletableFuture} returned by {@link AsyncResponseTransformer#prepare()} method
-     * in the first attempt of a request. This is used only for async streaming requests
+     * in the first attempt of a request. This used to be used only for async streaming requests.
+     *
+     * The purpose this attribute used to service has been replaced with
+     * {@link SdkInternalExecutionAttribute#EXECUTION_ATTEMPT} which now tracks the execution attempt number and is
+     * incremented with successive retries.
      */
+    @Deprecated
```

Comment:
I'd vote for removing it. It's a protected class and I'd be surprised if it's being used by any customers, and it's a pain to maintain it in the future.

Reply:
Yes already done actually. I refactored the new execution attribute into a truly internal class, and left the rest of this one alone as it was being referenced from the genned code.